hudi-common/src/main/java/org/apache/hudi/common/table/log/block/HoodieAvroDataBlock.java

```diff
@@ -115,13 +117,13 @@ protected byte[] serializeRecords(List<HoodieRecord> records) throws IOException
     output.writeInt(records.size());
 
     // 3. Write the records
-    for (HoodieRecord s : records) {
+    for (HoodieRecord<?> s : records) {
       ByteArrayOutputStream temp = new ByteArrayOutputStream();
       BinaryEncoder encoder = EncoderFactory.get().binaryEncoder(temp, encoderCache.get());
       encoderCache.set(encoder);
       try {
         // Encode the record into bytes
-        IndexedRecord data = (IndexedRecord) s.toIndexedRecord(schema, new Properties()).get();
+        IndexedRecord data = s.toIndexedRecord(schema, new Properties()).get();
```

Comment:
See my comment above regarding deprecation of `toIndexedRecord`: we shouldn't be using `toIndexedRecord` here -- instead we should cast the record to `HoodieAvroRecord` and access the payload directly